src/commands.ts

```diff
@@ -42,12 +44,51 @@ export default function registerCommands(commandObserver: CommandObserver, packa
 					});
 				});
 		}),
+		vscode.commands.registerCommand('pgproj.deploy.current', async (args) => {
+			requireDotNetSdk(dotNetSdkVersion).then(
+				async dotnet => {
+					await deployCurrentProject(args, dotnet, commandObserver);
+				});
+		}),
 		vscode.commands.registerCommand('pgproj.add.new', async (args) => {
-			await addNewPostgreSQLProject(args, packageInfo.maxSupportedPostgreSQLProjectSDK, commandObserver);
+			await addNewPostgreSQLProject(args, packageInfo.maxSupportedPostgreSQLProjectSDK);
 		})
 	];
 }
 
+async function deployCurrentProject(args, dotNetSdk: DotNetInfo, commandObserver:  CommandObserver, cancelToken?: vscode.CancellationToken) {
+	let project = '';
+	if (!args) {
+		const activeEditor = vscode.window.activeTextEditor;
+		if (activeEditor !== undefined) {
+			project = activeEditor.document.uri.fsPath;
+		}
+	} else {
+		project = args.fsPath;
+	}
+
+	await buildProjects(dotNetSdk, [project], commandObserver, cancelToken).then(async result => {
+		if (result && !result.some(b => b.status === buildStatus.Failure || b.status === buildStatus.Skipped)) {
+			var projectName = path.basename(project, '.pgproj');
+			var buildFiles = await vscode.workspace.findFiles('{**/'+ projectName +'.sql}');
+			if (buildFiles && buildFiles.length > 0) {
+				var connection = await azdata.connection.openConnectionDialog([Constants.providerId]);
+				if (connection) {
+					var fileName = buildFiles[0];
+					vscode.workspace.openTextDocument(fileName).then(doc => {
+						vscode.window.showTextDocument(doc, vscode.ViewColumn.Active, false).then(() => {
+							let filePath = doc.uri.toString();
+							azdata.queryeditor.connect(filePath, connection.connectionId).then(() => azdata.queryeditor.runQuery(filePath, undefined, false));
```

Comment:
How does it surface in the UI? Does it really spawn new editor window?
Since we are using this same extension to connect and run queries, can we just have an internal API to connect and run a query, by-passing ADS infrastructure?

Reply:
This will open up the project.sql file and deploy that and show errors below if any.

Reply:
That will be a breaking change though, as extension will expect the target to exist.., We probably should do something like that before the first release.